src/Pages/Signup/Signup.js

```diff
@@ -0,0 +1,294 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { SERVER_SIGNUP } from './Data/Config';
+import './Signup.scss';
+
+class Signup extends Component{
+  constructor(){
+    super();
+    this.state = {
+      username: "",
+      id: "",
+      password: "",
+      againpw: "",
+      phone: "",
+      birth: "",
+      name: "",
+      allChecked: false,
+      firstCheck: false,
+      secondCheck: false,
+      thirdCheck: false,
+    };
+  }
+
+  signupInput = (e) => {
+    const {id, value} = e.target;
+
+    this.setState({[id] : value})
+  }
+  
+  signupId = (e) => {
+    const { id, username, password, phone, birth} = this.state;
+
+    fetch(SERVER_SIGNUP, {
+      method: "POST",
+      body: JSON.stringify({
+        name: username,
+        email: id,
+        password: password,
+        phone_number: phone,
+        date_of_birth: birth,
+      })
+    })
+      .then(res => res.json())
+      .then(result => {
+        console.log(result);
+
+        if(result.error === "EXIST_EMAIL") {
+          alert('중복된 이메일 입니다.')
+          return;
+        }
+
+        if(result.error === "EXIST_PHONE_NUMBER") {
+          alert('사용중인 휴대폰 번호입니다.')
+          return;
+        }
+        if(result.message === "SUCCESS") {
+          alert('회원가입 성공!')
+          this.props.history.push('./main')
+        }
+      })
+  }
+
+  handleAllCheck = () =>{
+    const { allChecked } = this.state;
+    
+    this.setState({
+      firstCheck: !allChecked,
+      secondCheck: !allChecked,
+      thirdCheck: !allChecked,
+      allChecked: !allChecked,
+    }); 
+  };
+
+  handleCheck = e => {
+    const { name } = e.target;
+
+    this.setState({ [name] : !this.state.[name]})
+  }
```

Comment:
잘 줄여주셨는데 잘 되던가요...? ㅎ 점 하나를 빼셔야 할거 같습니다.
```suggestion
  handleCheck = e => {
    const { name } = e.target;

    this.setState({ [name] : !this.state[name]})
  }
```

Reply:
실행은 되는데 코드에선 오류가 떠서 찾고있었습니다.. 감사합니다!

Reply:
```suggestion
  handleCheck = e => {
    const { name } = e.target;

    this.setState({ [name + "check"] : !this.state[name + "check"]})
  }
```

Reply:
어제 종택님이 말씀해주신대로 네임값과 checked값을 같게하지않고 계산된속성명으로 변경하였습니다!
```suggestion
  handleCheck = e => {
    const { name } = e.target;

    this.setState({ [name + "Check"] : !this.state.[name + "Check"]})
  }
```

Reply:
@youngho052 ㅋㅋㅋ 이건 원리를 알려드리는 과정에서 이해를 돕기 위해 썼던 내용이었습니다. 아무튼 계산된 속성명 잘 활용해보세요!